### ISSUE TITLE
Response frame is too short

### ISSUE THREAD

**Issue by schottme** [catalinsanda/org.openhab.binding.solarman#34]
Hi all, 
I get following error if I connect the hybrid inventor with LSE (ethernet) with this binding. I use the 0.3.2 Version.
What can I do?

![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/143509593/ceacdf75-cf55-4e64-b275-40eab1d15bd8)

**Comment by catalinsanda** [comment]
Hello,

The response "A5 1700 1045 0300 F6 02" doesn't seem to follow the Solarman V5 protocol as described here: https://pysolarmanv5.readthedocs.io/en/stable/solarmanv5_protocol.html. This protocol was reverse engineered so no "official" documentation for it exists AFAIK. Have no idea what to do next and what the  0300 F6 02 error code means.

If you don't have the latest Logger firmware you can get in touch with the Deye support to ask for an update, maybe this will make it work.

**Comment by KeeperOf7Keys** [comment]
This is also the new Logger Firmware MW3_16U_5406_2.32-D1 of Deye Inverter which is used with the Relais extension. Inverter version data is like following communication protocol is version 0x0202 with HW version 0x0102 DC-Master Firmware is 0.2.3.6 with AC Version number 1.3.1.2
Compared to the ones which are working with the binding e.g. Logger Firmware MW3_16U_5406_1.53 and Inverter versions communication protocol 0x0201 HW Version 0x0102 DC_Master Firmware 0.1.1.7 with AC Version number 1.2.1.8

As it can be seen it is not an old Firmware - it is a new one. With different communication protocol.

It may be the reason mentioned for Solarman HomeAssistant binding: 
[https://github.com/StephanJoubert/home_assistant_solarman/discussions/271#discussioncomment-7158802](url)

With a Debugger Thing for Deye Inverters I can see Version and Serial numbers of inverter and Firmware version of logger.The other responses I could not evaluate yet.
The relais box is doing an automatic update to Logger version MW3_16U_5406_2.27

**Comment by catalinsanda** [comment]
In the provided link the issue seems to be "pysolarmanv5 is seeing a mixture of requested v5 frames and answers to AT-commands", which is something different that what we have here.

If indeed there is a newer and incompatible version of the Solarman v5, do you know of any library which was updated to support it, to "inspire" myself from their work?

**Comment by KeeperOf7Keys** [comment]
Up to now I only saw messages from users that complain read out of data is not working. My plan is to add a logger to collect the network traffic together with the Solarman app I hope to get the data analysed. But this will take some time.

**Comment by KeeperOf7Keys** [comment]
After checking and comparison of the already existing communication protocol there are quite some differences.
Request to get long data from inverter.
Data: a50a0010173c7ad315e9ed0001243c2166780000000515
after acknowledge and keep alive response with 343 Bytes is there
Data:
0000   98 9b cb a8 b9 72 40 2a 8f 38 00 12 08 00 45 00
0010   01 7f 29 36 00 00 ff 06 d0 7e c0 a8 b2 31 23 9d
0020   2a 4d 4e d9 27 10 00 00 64 28 82 c2 63 c8 50 18
0030   0c 42 f5 3f 00 00 a5 4a 01 10 42 00 7c d3 15 e9
0040   ed 01 08 54 9b ef 5a 00 5f 1c 00 00 ef 4c c6 65
0050   01 00 00 0f 00 00 xx xx xx xx xx xx xx xx xx xx
0060   02 00 d2 00 00 00 ae 07 00 00 00 00 00 00 10 09
0070   00 00 00 00 14 00 00 00 00 00 88 13 e8 01 00 00
0080   80 07 18 fc 18 fc 18 fc 18 fc 00 00 01 00 00 00
0090   00 00 00 00 00 00 3c 01 5d 00 47 01 42 00 00 00
00a0   00 00 00 00 00 00 56 30 2e 32 2e 30 2e 32 56 30
00b0   2e 32 2e 33 2e 36 56 31 2e 33 2e 32 2e 32 00 00
00c0   00 00 1f 40 02 01 00 04 00 0b 00 09 00 00 00 00
00d0   00 00 03 d6 00 00 03 7f 00 00 00 00 00 00 00 00
00e0   xx xx xx xx xx xx xx xx xx xx 2d 31 20 20 20 20  ident of solar panel 1
00f0   xx xx xx xx xx xx xx xx xx xx 2d 32 20 20 20 20  ident of solar panel 2
0100   xx xx xx xx xx xx xx xx xx xx 2d 33 20 20 20 20  ident of solar panel 3
0110   xx xx xx xx xx xx xx xx xx xx 2d 34 20 20 20 20  ident of solar panel 4
0120   13 ba 00 28 00 01 00 00 00 02 00 00 0b 3b 07 26
0130   14 1e 12 8e 00 3c 18 04 12 11 1b 37 00 01 00 01
0140   02 02 36 13 22 00 00 07 d0 00 00 00 00 00 00 00
0150   00 00 00 00 00 00 ff 09 e2 04 0b 00 01 13 ba 00
0160   28 00 00 13 ba 00 64 07 d0 00 64 00 00 00 00 00
0170   01 00 00 00 08 00 00 07 d0 03 e8 13 88 03 e8 1f
0180   40 fc 36 27 10 fc 4a 00 01 00 00 e6 15
There is some data ASCII coded - e.g. serial number of several data. xx is e.g. the serial number of inverter
Further already known data will be added the next days.
While using AT commands to read out registers is could be seen that the data is mainly at the same place like alreay analysed. There are some additional data with new protocol available.
In addtion there are 4 more packages - 2 with data length of 23 Bytes, 1 with 14 Bytes, another one with 311 Bytes which contains mainly data from external relay box. Details will follow.

**Comment by catalinsanda** [comment]
Can't tell for sure, but this looks like it's not encapsulating Modbus as usual, but something differently. Not much to work on.

**Comment by KeeperOf7Keys** [comment]
I have also send the AT commands with [deye-logger-at-cmd](https://github.com/s10l/deye-logger-at-cmd/tree/main). Do you like to see the responses also?

**Comment by catalinsanda** [comment]
Yes, sure, but I think those are part of a separate, control protocol.
